packages/gatsby/cache-dir/loader.js

```diff
@@ -381,6 +386,47 @@ export class ProdLoader extends BaseLoader {
         )
       })
   }
+
+  loadPageDataJson(rawPath) {
+    return super
+      .loadPageDataJson(rawPath)
+      .then(data => {
+        if (data.notFound) {
+          // check if html file exist using HEAD request:
+          // if it does we should navigate to it instead of showing 404
+          console.log(`[prod-loader] checking if html file exist`, rawPath, {
+            ...data,
+          })
+
+          return doFetch(rawPath, `HEAD`).then(req => {
+            console.log(
+              `[prod-loader] html check result`,
+              rawPath,
+              { status: req.status },
+              { ...req }
+            )
+            if (req.status === 200) {
+              // page actually exist (or we asked for 404 )
+              return {
+                status: `error`,
+              }
+            } else if (req.status === 404) {
+              // page truly doesn't exist
+              return data
+            } else {
+              console.log(`[prod-loader] something else???`, req.status)
```

Comment:
in json loading code we only handle code 200, 404 ... and 500 - do we need to handle more in HEAD request check than 200 and 404?